### ISSUE TITLE
Search feature on documentation

### ISSUE THREAD

**Issue by Hans5958** [ScratchAddons/website-v2#168]
This is a dream of mine, but I haven't got time or skill to do it, so if anyone can implement a search feature on the docs, I will appreciate it.

Preferably without Algolia or other third-party services

Some resources ideas to start with (go ahead if you have other ideas):
- https://fusejs.io/
- https://scratchaddons.com/index.xml
- https://scratchaddons.com/en/sitemap.xml

**Comment by RedGuy12** [comment]
fuse is actually horrible tbh, the results are not accurate at all

see, this search returns nothing because i didn't use double quotes
![image](https://user-images.githubusercontent.com/61329810/156901594-2d9c0193-2dd2-4118-b944-e9f1c713aff6.png)

**Comment by mxmou** [comment]
> see, this search returns nothing because i didn't use double quotes

A single quote means "include-match" according to [the documentation](https://fusejs.io/examples.html#extended-search), but I don't know how that's different from just typing the word.

A more serious problem is that even if extended search is enabled (without it, you have to enter the words in the same order as they appear in the result), you can only search individual fields.
![image](https://user-images.githubusercontent.com/51849865/156912685-13f79fbf-e3c7-4007-844d-69c7b77ff2dd.png)
![image](https://user-images.githubusercontent.com/51849865/156912695-44ee941b-0b28-407e-9777-53dabf49f4e5.png)
![image](https://user-images.githubusercontent.com/51849865/156912700-01c94c3a-9add-4f91-860f-54279e437031.png)

For documentation search, that would mean "joining volunteer localization" would return no results because "joining" only appears in the title and "volunteer" only appears in the page content.

**Comment by Hans5958** [comment]
Yeah, it was just my random idea based on the current implementation on Scratch Addons. If anyone knows better libraries then go ahead.

**Comment by Hans5958** [comment]
I forgot this is existed:

- https://www.npmjs.com/package/hugo-lunr
- https://gohugo.io/tools/search/

**Comment by Weredime** [comment]
> Preferably without Algolia

:(
